crates/qasm2/src/expr.rs

```diff
@@ -502,7 +502,13 @@ impl<'a> ExprParser<'a> {
             | TokenType::Sqrt
             | TokenType::Tan => Ok(Some(Atom::Function(token.ttype.into()))),
             TokenType::Real => Ok(Some(Atom::Const(token.real(self.context)))),
-            TokenType::Integer => Ok(Some(Atom::Const(token.int(self.context) as f64))),
+            // This deliberately parses an _integer_ token as a float, since all OpenQASM 2.0
+            // integers can be interpreted as floats, and doing that allows us to gracefully handle
+            // cases where a huge float would overflow a `usize`.  Never mind that in such a case,
+            // there's almost certainly precision loss from the floating-point representating
+            // having insufficient mantissa digits to faithfully represent the angle mod 2pi;
+            // that's not our fault in the parser.
+            TokenType::Integer => Ok(Some(Atom::Const(token.real(self.context)))),
```

Comment:
Nitpicking, but would it be better to use a single case for `Real | Integer` so it's more immediately obvious to the reader that these are handled in exactly the same way?

The comment ought to stay, though 🙂.

Reply:
Good call, done in 9f4fbf1c67.